### PR TITLE
improve: replace type traits with STL

### PR DESCRIPTION
**Pull request by ranisalt**
# Description

Since C++20 is enabled, these type traits are included in the standard library and don't need to be implemented here.

## How Has This Been Tested

  - [ ] It compiles
  - [ ] Test B

## Checklist

  - [x] My code follows the style guidelines of this project
  - [x] I have performed a self-review of my own code
  - [x] I checked the PR checks reports
  - [ ] I have commented my code, particularly in hard-to-understand areas
  - [ ] I have made corresponding changes to the documentation
  - [x] My changes generate no new warnings
  - [ ] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by ranisalt** [comment]
Looks like Windows builds are broken, they're failing even on master

File: src/framework/stdext/format.h [hunk 1]
```diff
@@ -74,7 +72,7 @@ namespace stdext
     template<typename... Args>
     int snprintf(char* s, size_t maxlen, const char* format, const Args&... args)
     {
-        std::tuple<typename replace_extent<Args>::type...> tuple(args...);
+        std::tuple<std::decay_t<Args>...> tuple(args...);
```
Comment (mehah) [inline src/framework/stdext/format.h:76]:
```
1>D:\programacao\jogos\tibia\otclient\src\framework\stdext\format.h(75,48): error C2661: 'std::tuple<int,char *,CHAR *,unsigned __int64,unsigned __int64 >::tuple': no ​​overloaded function takes 5 arguments
1>D:\programacao\jogos\tibia\otclient\src\framework\stdext\format.h(75,48): error C2661: 'std::tuple<int,char *,unsigned __int64>::tuple': no overloaded function takes 3 arguments
1>D:\programacao\jogos\tibia\otclient\src\framework\stdext\format.h(75,48): error C2665: 'std::tuple<char *>::tuple': no ​​overloaded function can convert all types of arguments
```

possible fix: (put condition to not compile in MSVC)
```c++
#ifndef _MSC_VER
    // Improved snprintf that accepts std::string and other types
    template<typename... Args>
    int snprintf(char* s, size_t maxlen, const char* format, const Args&... args)
    {
        std::tuple<std::decay_t<Args>...> tuple(args...);
        return expand_snprintf<std::tuple_size_v<decltype(tuple)>>::call(s, maxlen, format, tuple);
    }

    template<typename... Args>
    int snprintf(char* s, size_t maxlen, const char* format)
    {
        std::strncpy(s, format, maxlen);
        s[maxlen - 1] = 0;
        return strlen(s);
    }
#endif
```

**Comment by mehah** [comment]
> Looks like Windows builds are broken, they're failing even on master

we will analyze, but for now we can ignore this error and merge, just set this condition to not compile in MSVC.
thx

File: src/framework/stdext/format.h (hunk 1 above)
Comment (mehah) [reply src/framework/stdext/format.h:76]: @ranisalt 
I did the review and forgot to apply it. :sweat_smile:

**Comment by mehah** [comment]
@ranisalt 
Unfortunately, after these changes, the modules are showing errors. :S

**Pull request merged**